app/routes/events/components/StripeElement.js

```diff
@@ -4,38 +4,146 @@ import {
+  message: string,
+  doc_url: string
+};
+
+type FormProps = Props & {
+  fontSize?: number
+};
+
+type _SplitFormProps = FormProps & {
+  ledgend: string,
+  setError: Object => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  stripe: Stripe
+};
+
+type _PaymentRequestFormProps = FormProps & {
+  setError: Object => void,
+  setSuccess: () => void,
+  setLoading: boolean => void,
+  setPaymentRequest: boolean => void,
+  stripe: Stripe
+};
+
+type FormState = {
+  error?: StripeError | null,
+  success?: boolean,
+  loading: boolean,
+  paymentRequest: boolean
+};
 
 type State = {
-  paymentRequest: Object,
+  paymentRequest?: Object,
   canMakePayment?: boolean
 };
+const createOptions = (fontSize, padding) => {
+  return {
+    style: {
+      base: {
+        fontSize,
+        color: '#424770',
+        letterSpacing: '0.025em',
+        fontFamily: 'Source Code Pro, monospace',
+        '::placeholder': {
+          color: '#aab7c4'
+        },
+        padding
+      },
+      invalid: {
+        color: '#9e2146'
+      }
+    }
+  };
+};
+class _SplitForm extends React.Component<_SplitFormProps, FormState> {
+  handleSubmit = async ev => {
+    ev.preventDefault();
+    const { stripe, createPaymentIntent } = this.props;
+    if (stripe) {
+      const { payload } = await createPaymentIntent();
+      this.props.setLoading(true);
+      const { error } = await stripe.handleCardPayment(payload.clientSecret);
+      if (error) {
+        this.props.setError(error);
+      } else {
+        this.props.setSuccess();
+      }
+      this.props.setLoading(false);
+    }
+  };
+  render() {
+    return (
+      <form style={{ width: '100%' }} onSubmit={this.handleSubmit}>
+        <fieldset className={stripeStyles.elementsFieldset}>
+          <legend className={stripeStyles.elementsLedgend}>
+            {this.props.ledgend}
+          </legend>
+          <label className={stripeStyles.StripeLabel}>
+            Kortnummer
+            <CardNumberElement
+              className={stripeStyles.StripeElement}
+              {...createOptions(this.props.fontSize)}
+            />
+          </label>
+          <label className={stripeStyles.StripeLabel}>
+            Utløpsdato
+            <CardExpiryElement
+              className={stripeStyles.StripeElement}
+              {...createOptions(this.props.fontSize)}
+            />
+          </label>
+          <label className={stripeStyles.StripeLabel}>
+            CVC
+            <CardCVCElement
+              className={stripeStyles.StripeElement}
+              onReady={() => this.setState({})}
+              {...createOptions(this.props.fontSize)}
+            />
+          </label>
+          <button className={stripeStyles.StripeButton}>Betal</button>
+        </fieldset>
+      </form>
+    );
+  }
+}
 
-const paymentButtonWidth = 130;
-
-class PaymentRequestForm extends React.Component<FormProps, State> {
+class _PaymentRequestForm extends React.Component<
```

Comment:
Is there a particular reason for all these `_Thing`-things? Seems a bit strange to me.

Reply:
Yeah, I'll improve the naming in general in this file